include/matx/transforms/cub.h

```diff
@@ -437,107 +437,40 @@ class matxCubPlan_t {
 #ifdef __CUDACC__
   if constexpr (is_tensor_view_v<InputOperator>)
   {
-
-    //////////////////////////////////////////////////////
-    //////           Rank 1 Tensors               ////////
-    //////////////////////////////////////////////////////
-    if (RANK == 1)
-    {
-      if (dir == SORT_DIR_ASC)
-      {
-        cub::DeviceSegmentedSort::SortKeys(
-            d_temp, temp_storage_bytes, a.Data(), a_out.Data(),
-            static_cast<int>(a.Size(RANK-1)),
-            1,
-            BeginOffset{a}, EndOffset{a},
-            stream);
-      }
-      else
-      {
-
-        cub::DeviceSegmentedSort::SortKeysDescending(
-            d_temp, temp_storage_bytes, a.Data(), a_out.Data(),
-            static_cast<int>(a.Size(RANK-1)),
-            1,
-            BeginOffset{a}, EndOffset{a},
-            stream);
-      }
+     // CUB added support for large items and segments in https://github.com/NVIDIA/cccl/pull/3308
+    int64_t _num_segments = 1;
+    for (int i = 0; i < RANK - 1; i++) {
+      _num_segments *= a.Size(i);
     }
-
-    //////////////////////////////////////////////////////
-    //////           Rank 2 Tensors               ////////
-    //////////////////////////////////////////////////////
-    else if (RANK == 2 || d_temp == nullptr)
+    int64_t _num_items = _num_segments * a.Size(RANK - 1);
+#if 0  // TODO: add conditional on CUB_MAJOR_VERSION once released
```

Comment:
What is the conditional you need here? Which CUB version supports this feature?

Reply:
I don't think it's supported in a release version yet (I don't see it mentioned in https://github.com/NVIDIA/cccl/releases), but once it has we can update it.